core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -131,6 +131,11 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     set("spark.home", home)
   }
 
+  /** Set the metrics configuration property */
+  def setMetricsProperty(name: String, value: String): SparkConf = {
```

Comment:
I'm a little unsure about adding these metric-specific methods here. They're very particular to a specific scenario, while SparkConf is supposed to be a generic configuration class. I'd rather not add these (since I expect most people to set these via config files anyway).